src/vm/datatypes/strings.c

```diff
@@ -209,16 +234,22 @@ static Value splitString(DictuVM *vm, int argCount, Value *args) {
 
     pop(vm);
 
-    FREE_ARRAY(vm, char, tmpFree, string->length + 1);
     return OBJ_VAL(list);
 }
 
 static Value containsString(DictuVM *vm, int argCount, Value *args) {
+
     if (argCount != 1) {
         runtimeError(vm, "contains() takes 1 argument (%d given)", argCount);
         return EMPTY_VAL;
     }
-
+    {
+        ObjString *string = AS_STRING(args[0]);
```

Comment:
If we're checking utf8 legality would we also need to check the needle (args[1])?